plugins/google-sheets/src/pages/MapSheetFields.tsx

```diff
@@ -72,11 +72,17 @@ const inferFieldType = (cellValue: CellValue): CollectionFieldType => {
 
         if (/^#[0-9A-Fa-f]{6}$/.test(cellValue)) return "color"
 
-        if (/\.(jpeg|jpg|gif|png)$/.test(cellValue)) return "image"
+        if (/<[a-z][\s\S]*>/i.test(cellValue)) return "formattedText"
 
-        if (/^(https?:\/\/)?([a-zA-Z0-9.-]+)\.([a-zA-Z]{2,6})([/\w .-]*)*\/?$/.test(cellValue)) return "link"
+        try {
+            new URL(cellValue)
 
-        if (/<[a-z][\s\S]*>/i.test(cellValue)) return "formattedText"
+            if (/\.(jpeg|jpg|gif|png)$/.test(cellValue)) return "image"
```

Comment:
Can you, please, update this to also include all supported file types:

```
["gif", "image/gif"],
["jpg", "image/jpeg"],
["jpeg", "image/jpeg"],
["png", "image/png"],
["apng", "image/png"],
["svg", "image/svg+xml"],
["webp", "image/webp"],
```

And make it case-insensitive? Right now it rejects `foo.JPG`, for example. Might be cleaner to just do a `const cellValueLower = cellValue.toLowerCase()` at the beginning of the block and update the regular expressions to use that, without any upper ranges like `A-F` or `i` flags.

Reply:
Yes, that's indeed cleaner! Thanks for flagging 😁